src/lib/io/sol-iio.c

```diff
@@ -746,13 +803,17 @@ sol_iio_open(int device_id, const struct sol_iio_config *config)
         device->reader_cb_data = config->data;
 
         if (config->trigger_name && config->trigger_name[0] != '\0') {
-            if (!set_current_trigger(device, config->trigger_name)) {
-                SOL_WRN("Could not set device%d current trigger",
-                    device->device_id);
+            /* if name start with HRTIMER_TRIGGER_PREFIX, it is a hrtimer trigger */
+            if (strstartswith(config->trigger_name, HRTIMER_TRIGGER_PREFIX)) {
+                if (!set_current_trigger(device, config->trigger_name + strlen(HRTIMER_TRIGGER_PREFIX)))
```

Comment:
Based on my test, writing a null string to the current_trigger does not break the logic. For safe I add a null detect